src/mongo/db/storage/mmap_v1/mmap_windows.cpp

```diff
@@ -184,8 +184,8 @@ void* MemoryMappedFile::map(const char* filenameIn, unsigned long long& length)
     strncpy(filename, filenameIn, 255);
     filename[255] = 0;
     {
-        size_t len = strlen(filename);
-        for (size_t i = len - 1; i >= 0; i--) {
+        int len = strlen(filename);
```

Comment:
@ZharkovIlia If you would like to see this PR merged (and, I hope you do, because you put a lot of work in to it!), we will still need to sort out this one last issue. Please let me know if I can help clarify anything in this discussion.